src/base/static/components/story-sidebar/index.js

```diff
@@ -0,0 +1,190 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { Map, OrderedMap, fromJS } from "immutable";
+
+import StoryChapter from "./story-chapter";
+import constants from "../../constants";
+
+import "./index.scss";
+
+import { story as storyConfig, place as placeConfig } from "config";
+
+class StorySidebar extends Component {
+  constructor(props) {
+    super(props);
+
+    this.stories = Object.keys(storyConfig).reduce((memo, storyName) => {
+      return memo.set(
+        storyName,
+        Object.keys(storyConfig[storyName].order).reduce((memo, chapterUrl) => {
+          return memo.set(
+            chapterUrl,
+            fromJS(this.getModelFromUrl(chapterUrl).attributes),
+          );
+        }, OrderedMap()),
+      );
+    }, OrderedMap());
+
+    this.state = {
+      currentStoryHeader: "",
+      currentStoryDescription: "",
+      currentStoryName: "",
+      currentStoryChapters: OrderedMap(),
+      currentStoryRoute: "",
+      currentChapter: Map(),
+    };
+  }
+
+  componentWillMount() {
+    this.isInitialized = false;
+    this.loadAndSetStoryChapter(Backbone.history.getFragment().split("/"));
+    this.isInitialized = true;
+
+    this.props.router.on("route", (fn, route) => {
+      this.loadAndSetStoryChapter(route);
+    });
+  }
+
+  loadAndSetStoryChapter(route) {
+    let isRouteFound = false;
+    const joinedRoute = route.join("/");
+    this.stories.forEach((story, storyName) => {
+      if (story.get(joinedRoute)) {
+        // If the current route exists in this story, load the story and set
+        // the current chapter and name.
+        isRouteFound = true;
+        this.setState({
+          currentStoryHeader: storyConfig[storyName].header,
+          currentStoryDescription: storyConfig[storyName].description,
+          currentStoryName: storyName,
+          currentStoryChapters: story,
+          currentStoryRoute: joinedRoute,
+          currentChapter: story.get(joinedRoute),
+        });
+      }
+    });
+
+    // If we can't find a match for the current route and we haven't
+    // initialized the story sidebar, load content from the first story
+    // provided in the config so the sidebar is never without content.
+    if (!isRouteFound) {
+      this.setState({
+        currentStoryHeader: !this.isInitialized
+          ? storyConfig[this.stories.keySeq().first()].header
+          : "",
+        currentStoryDescription: !this.stories.isInitialized
+          ? storyConfig[this.stories.keySeq().first()].description
+          : "",
+        currentStoryName: !this.isInitialized
+          ? this.stories.keySeq().first()
+          : this.state.currentStoryName,
+        currentStoryChapters: !this.isInitialized
+          ? this.stories.first()
+          : this.state.currentStoryChapters,
+        currentStoryRoute: "",
+        currentChapter: Map(),
+      });
+    }
+  }
+
+  getModelFromUrl(url) {
```

Comment:
nit: There is some messy logic going on here. I'd prefer to make it a pure function by having it accept a `places` arg, and possibly refactoring it out into our util methods.

That would make it easier to unit test and refactor when we refactor out our Backbone models.

Reply:
Factoring this out to the utils sounds like a good idea.